hail/python/hail/table.py

```diff
@@ -304,22 +304,31 @@ class Table(ExprContainer):
     """
 
     @staticmethod
-    def _from_ir(table_ir):
-        jir = Env.hail().expr.Parser.parse_table_ir(str(table_ir))
-        return Table(Env.hail().table.Table(Env.hc()._jhc, jir))
+    def _from_java(jt):
+        return Table(JavaTable(jt.tir()))
 
-    def __init__(self, jt):
+    def __init__(self, tir):
         super(Table, self).__init__()
 
-        self._jt = jt
+        self._tir = tir
+
+        r = Renderer(stop_at_jir=True)
```

Comment:
This is almost perfect, one last minor suggestion: add a `BaseIR.to_java_ir` that returns a Java IR.  It should look something like:

```
  def to_java_ir():
    if not hasattr(self, '_jir'):
      r = Renderer(stop_at_jir=True)
      ...
      self._jir = self.parse(code, {}, ir_map)
  return self._jir
```

Where parse is defined on IR, TableIR and MatrixIR and calls the right Scala parser function.  Reasonable?